binding.c

```diff
@@ -11,24 +99,2109 @@ napi_value sn_randombytes_random (napi_env env, napi_callback_info info) {
+
+  return NULL;
+}
+
+napi_value sn_crypto_core_ed25519_scalar_negate (napi_env env, napi_callback_info info) {
+  SN_ARGV(2, crypto_core_ed25519_scalar_negate)
+
+  SN_ARGV_TYPEDARRAY(neg, 0)
+  SN_ARGV_TYPEDARRAY(s, 1)
+
+  SN_THROWS(neg_size != crypto_core_ed25519_SCALARBYTES, "negative buffer must be 32 bytes")
+  SN_THROWS(s_size != crypto_core_ed25519_SCALARBYTES, "scalar must be 32 bytes")
+
+  crypto_core_ed25519_scalar_negate(neg_data, s_data);
+
+  return NULL;
+}
+
+napi_value sn_crypto_core_ed25519_scalar_complement (napi_env env, napi_callback_info info) {
+  SN_ARGV(2, crypto_core_ed25519_scalar_complement)
+
+  SN_ARGV_TYPEDARRAY(comp, 0)
+  SN_ARGV_TYPEDARRAY(s, 1)
+
+  SN_THROWS(comp_size != crypto_core_ed25519_SCALARBYTES, "complement buffer must be 32 bytes")
+  SN_THROWS(s_size != crypto_core_ed25519_SCALARBYTES, "scalar must be 32 bytes")
+
+  crypto_core_ed25519_scalar_complement(comp_data, s_data);
+
+  return NULL;
+}
+
+napi_value sn_crypto_core_ed25519_scalar_add (napi_env env, napi_callback_info info) {
+  SN_ARGV(3, crypto_core_ed25519_scalar_add)
+
+  SN_ARGV_TYPEDARRAY(z, 0)
+  SN_ARGV_TYPEDARRAY(x, 1)
+  SN_ARGV_TYPEDARRAY(y, 2)
+
+  SN_THROWS(z_size != crypto_core_ed25519_SCALARBYTES, "result buffer must be 32 bytes")
+  SN_THROWS(x_size != crypto_core_ed25519_SCALARBYTES, "x must be 32 bytes")
+  SN_THROWS(y_size != crypto_core_ed25519_SCALARBYTES, "y must be 32 bytes")
+
+  crypto_core_ed25519_scalar_add(z_data, x_data, y_data);
+
+  return NULL;
+}
+
+
+napi_value sn_crypto_core_ed25519_scalar_sub (napi_env env, napi_callback_info info) {
+  SN_ARGV(3, crypto_core_ed25519_scalar_sub)
+
+  SN_ARGV_TYPEDARRAY(z, 0)
+  SN_ARGV_TYPEDARRAY(x, 1)
+  SN_ARGV_TYPEDARRAY(y, 2)
+
+  SN_THROWS(z_size != crypto_core_ed25519_SCALARBYTES, "result buffer must be 32 bytes")
+  SN_THROWS(x_size != crypto_core_ed25519_SCALARBYTES, "x must be 32 bytes")
+  SN_THROWS(y_size != crypto_core_ed25519_SCALARBYTES, "y must be 32 bytes")
+
+  crypto_core_ed25519_scalar_sub(z_data, x_data, y_data);
+
+  return NULL;
+}
+
+napi_value sn_crypto_shorthash (napi_env env, napi_callback_info info) {
+  SN_ARGV(3, crypto_shorthash)
+
+  SN_ARGV_TYPEDARRAY(output, 0)
+  SN_ARGV_TYPEDARRAY(input, 1)
+  SN_ARGV_TYPEDARRAY(key, 2)
+
+  SN_THROWS(output_size != crypto_shorthash_BYTES, "output must be 8 bytes")
+  SN_THROWS(key_size != crypto_shorthash_KEYBYTES, "key must be 16 bytes")
+
+  SN_RETURN(crypto_shorthash(output_data, input_data, input_size, key_data), "could not compute hash")
+}
+
+napi_value sn_crypto_kdf_keygen (napi_env env, napi_callback_info info) {
+  SN_ARGV(1, crypto_kdf_keygen)
+
+  SN_ARGV_TYPEDARRAY(key, 0)
+
+  SN_THROWS(key_size != crypto_kdf_KEYBYTES, "output must be 32 bytes")
+
+  crypto_kdf_keygen(key_data);
+
+  return NULL;
+}
+
+napi_value sn_crypto_kdf_derive_from_key (napi_env env, napi_callback_info info) {
+  SN_ARGV(4, crypto_kdf_derive_from_key)
+
+  SN_ARGV_TYPEDARRAY(subkey, 0)
+  SN_ARGV_UINT64(subkey_id, 1)
```

Comment:
should prob also be a 32bit uint, cc @emilbayes

Reply:
fixed by 3e2ae75